content/en/platform/corda/5.0/developing-applications/getting-started/utxo-advanced-ledger-extensions/advanced-design.md

```diff
@@ -0,0 +1,413 @@
+
+```kotlin
+public final class Create extends FungibleContractCreateCommand<ExampleFungibleState> {
+  
+   @NotNull
+   
+   public Class<ExampleFungibleState> getContractStateType() {
+   return ExampleFungibleState.class;
+  }
+  
+   @Override
+  
+   protected void onVerify(@NotNull final UtxoLedgerTransaction transaction) {
+    
+    // Verify additional Create constraints
+    
+  }
+}
+```
+
+The `FungibleContractUpdateCommand` supports updating existing fungible states and will verify the following constraints:
+
+* On fungible state(s) updating, at least one fungible state must be consumed.
+* On fungible state(s) updating, at least one fungible state must be created.
+* On fungible state(s) updating, the quantity of every created fungible state must be greater than zero.
+* On fungible state(s) updating, the sum of the unscaled values of the consumed states must be equal to the sum of the unscaled values of the created states.
+* On fungible state(s) updating, the sum of the consumed states that are fungible with each other must be equal to the sum of the created states that are fungible with each other.
+
+```kotlin
+public final class Update extends FungibleContractUpdateCommand<ExampleFungibleState> {
+  
+   @NotNull
+   
+    public Class<ExampleFungibleState> getContractStateType() {
+    return ExampleFungibleState.class;
+  }
+  
+   @Override
+  
+    protected void onVerify(@NotNull final UtxoLedgerTransaction transaction) {
+    
+    // Verify additional Update constraints
+    
+  }
+}
+```
+
+The `FungibleContractDeleteCommand` supports deleting existing fungible states and will verify the following constraints:
+
+* On fungible state(s) deleting, at least one fungible state input must be consumed.
+* On fungible state(s) deleting, the sum of the unscaled values of the consumed states must be greater than the sum of the unscaled values of the created states.
+* On fungible state(s) deleting, the sum of consumed states that are fungible with each other must be greater than the sum of the created states that are fungible with each other.
+
+```kotlin
+public final class Delete extends FungibleContractDeleteCommand<ExampleFungibleState> {
+  
+   @NotNull
+   
+    public Class<ExampleFungibleState> getContractStateType() {
+    return ExampleFungibleState.class;
+  }
+  
+   @Override
+  
+    protected void onVerify(@NotNull final UtxoLedgerTransaction transaction) {
+    
+    // Verify additional Delete constraints
+    
+  }
+}
+```
+
+## Designing a Fungible Contract
+
+A fungible contract can be implemented by extending the `FungibleContract` class, for example:
+
+```kotlin
+
+   public final class ExampleFungibleContract extends FungibleContract {
+  
+   @Override
+   
+   public List<Class<? extends FungibleContractCommand<?>>> getPermittedCommandTypes() {
+   return List.of(Create.class, Update.class, Delete.class);
+   
+  }
+}
+```
```

Comment:
```suggestion
```kotlin
public final class ExampleFungibleContract extends FungibleContract {
  @Override
  public List<Class<? extends FungibleContractCommand<?>>> getPermittedCommandTypes() {
    return List.of(Create.class, Update.class, Delete.class);
  }
}
```
```